arch/arm/core/aarch32/prep_c.c

```diff
@@ -59,6 +65,7 @@ static inline void relocate_vector_table(void)
 
 static inline void relocate_vector_table(void)
 {
+	set_low_vector();
```

Comment:
I do not think this makes sense. The `VECTOR_ADDRESS` may very well be HIVECS.

The current convention is to let the SoC init code set the SCTLR.V, as seen here:
https://github.com/zephyrproject-rtos/zephyr/blob/a0380aabff80c2476c7cdd6bc0844310ce4298ce/soc/arm/xilinx_zynqmp/soc.c#L33-L42

This could potentially be moved to a generalised implementation with a Kconfig to control it though.

Reply:
> So if we are setting VBAR, are we not assuming low vecs?

You are right. The real problem seems to be that `relocate_vector_table` is unconditionally called, so it does not look like it can support keeping the HIVECS after the initial boot anyway.